src/commands/Animation/anime.ts

```diff
@@ -1,122 +1,134 @@
-import { envIsDefined } from '#lib/env';
 import { LanguageKeys } from '#lib/i18n/languageKeys';
 import { PaginatedMessageCommand, SkyraPaginatedMessage } from '#lib/structures';
 import type { GuildMessage } from '#lib/types';
-import type { Kitsu } from '#lib/types/definitions/Kitsu';
+import { CdnUrls } from '#lib/types/Constants';
+import { fetchAniList, getAnime, parseDescription } from '#utils/APIs/AniList';
 import { sendLoadingMessage } from '#utils/util';
 import { ApplyOptions } from '@sapphire/decorators';
-import { fetch, FetchMethods, FetchResultTypes } from '@sapphire/fetch';
-import { MimeTypes } from '@sapphire/plugin-api';
-import { cutText } from '@sapphire/utilities';
-import { MessageEmbed } from 'discord.js';
-import type { TFunction } from 'i18next';
-import { stringify } from 'querystring';
-
-const API_URL = `https://${process.env.KITSU_ID}-dsn.algolia.net/1/indexes/production_media/query`;
+import { Time } from '@sapphire/time-utilities';
+import { filterNullish, isNullish } from '@sapphire/utilities';
+import { MessageEmbed, TextChannel } from 'discord.js';
 
 @ApplyOptions<PaginatedMessageCommand.Options>({
-	enabled: envIsDefined('KITSU_ID', 'KITSU_TOKEN'),
+	aliases: ['ani-list'],
 	cooldown: 10,
-	description: LanguageKeys.Commands.Animation.AnimeDescription,
-	extendedHelp: LanguageKeys.Commands.Animation.AnimeExtended
+	description: LanguageKeys.Commands.Animation.AniListAnimeDescription,
+	extendedHelp: LanguageKeys.Commands.Animation.AniListAnimeExtended
 })
 export class UserPaginatedMessageCommand extends PaginatedMessageCommand {
 	public async run(message: GuildMessage, args: PaginatedMessageCommand.Args) {
-		const animeName = await args.rest('string');
-		const response = await sendLoadingMessage(message, args.t);
+		const { t } = args;
+		const [search, loadingMessage] = await Promise.all([args.rest('string'), sendLoadingMessage(message, t)]);
```

Comment:
You seriously have no idea how much I hate this.

`args.rest('string')` is basically zero cost, it takes sub-ms to run, but regardless of whether it succeeds or not, you're always sending a message at potentially the same time `args.rest` could send another with a rejection.